### PR TITLE
#157731627 Allow managers delete or deactivate trainers 

### PR DESCRIPTION
**Pull request by hariclerry**
### What does this PR do?
This PR allows managers to delete or deactivate trainers
#### Description of Task to be completed?
Currently a manager cannot delete or deactivate trainers from a gym. They should be able to do this
### How should this be manually tested?

1. Follow the setup instructions in [README.rst](https://github.com/andela/wg-rogue-one/blob/ft-allow-managers-delete/deactivate-trainers-157731627/README.rst)
2. Login as an admin(Use default Username: admin, Password: admin)
3. Create a new gym and give it a name
4. Add a manager for the gym(copy the manager's username and password for Login)
5. Add trainers to the gym
6. Log out from the admin's account and login as the newly created manager
7. From the lists of users listed under the particular gym, click deactivate to deactivate their membership or delete to remove the trainer's account from the database.
8. You can activate a deactivated trainer too

### Screen Shots

Gym manager dashboard before deleting a trainer or deactivating a trainer

dashboard

Page showing delete and activate buttons

during

Gym manager dashboard after deactivating a trainer

deactivate

Gym manager dashboard after deleting a trainer

delete

### Any background context you want to provide?
Before adding this new feature, the admin was the only one authorized to delete and deactivate trainers. By adding this new feature into the app, gym managers now have permission to delete and deactivate trainers.

### What are the relevant pivotal tracker stories?
[#157731627](https://www.pivotaltracker.com/story/show/157731627)

### CODE REVIEW

**Comment by hariclerry** [comment]
Hey guys, can you please review my PR and let me know what you think. Am still fixing a few issues though like creating a drop-down button to display the Delete and Deactivate options.

**Review by King-Benx — CHANGES_REQUESTED** [review]
Your requirements.txt has deprecated packages such as the psycopg2-binary. Remove one of them.

Coverage is also repeated in requirements_devl.txt and requirements.txt. Remove the one in the requirements.txt

In your settings.py, try to match the settings that are in the develop branch at the time you send your commit as well as avoid leaving confidential information in the settings

After running your autopep8 or flek8, endeavour to go back and add a new line at the end of the file

**Comment by hariclerry** [comment]
@King-Benx, I have fixed those issues.

File: wger/config/tests/test_gym_config.py [hunk 1]
```diff
@@ -90,4 +90,4 @@ def test_update_userprofile(self):
         self.assertEqual(UserProfile.objects.filter(gym=gym).count(), 24)
 
         # 13 non-managers
-        self.assertEqual(GymUserConfig.objects.filter(gym=gym).count(), 13)
+        self.assertEqual(GymUserConfig.objects.filter(gym=gym).count(), 18)
```
Comment (emugaya) [inline wger/config/tests/test_gym_config.py]: The Count should match the Comment above. `18`

File: wger/core/views/user.py [hunk 2]
```diff
@@ -131,6 +134,126 @@ def delete(request, user_pk=None):
...
+                return HttpResponseRedirect(reverse('gym:gym:user-list', kwargs={'pk': gym_pk}))
+    context = {'form': form,
+               'user_delete': user,
+               'form_action': form_action}
+
+    return render(request, 'user/delete_account.html', context)
+
+
+@login_required
+def deactivate_trainer(request, user_pk=None, gym_pk=None):
```
Comment (emugaya) [inline wger/core/views/user.py]: Change method name to `deactivate_trainer_from_gym`

**Review by emugaya — CHANGES_REQUESTED** [review]
Refactor your code to have only one migration that adds the new fields.

File: wger/config/tests/test_gym_config.py (hunk 1 above)
Comment (hariclerry) [reply wger/config/tests/test_gym_config.py]: Okay. I have fixed that.

**Pull request merged**